osclib/core.py

```diff
@@ -288,7 +289,7 @@ def project_pseudometa_file_load(apiurl, project, filename, revision=None):
     project, package = project_pseudometa_package(apiurl, project)
     source_file = source_file_load(apiurl, project, package, filename, revision)
     if source_file is not None:
-        source_file = source_file.decode('utf-8').rstrip()
+        source_file = source_file.encode('utf-8').decode('utf-8').rstrip()
```

Comment:
lol, python drunk?

Reply:
`str` in python3 is a Unicode object, `decode()` isn't an attribute of that object anyway, str->bytes->str is the way force it to converting the encoding. *However* in python3 the default encoding should be utf-8 if not defining `codeing` in the py file, so I think we should remove most of `decode()` cases in our code?

Reply:
if source_file_load is now a str - then just don't do nothing, but rstrip() on it.

Reply:
at least I can't follow the reasoning to have an encode() there

Reply:
because I'm not sure what is `decode('utf-8')` suppose to do before.

Reply:
decode('utf-8') is converting byte to str - just the `decode_it` function does. It now breaks as you change the returned class of source_file_load, so removing it is the right thing to do

Reply:
yes, as I explained above. if `'utf-8'` is not a problem here.

Reply:
changed. if it has non-utf8 character breaks things we can revisit it here.

Reply:
that's for decode_it to handle :)

Reply:
yeah, but my understanding was deocde_it isn't mandatory to return a utf8 decoding. :)